src/cdk/scrolling/virtual-scroll-viewport.spec.ts

```diff
@@ -928,6 +928,31 @@ describe('CdkVirtualScrollViewport', () => {
       expect(testComponent.trackBy).toHaveBeenCalled();
     }));
   });
+
+  describe('with append only', () => {
+    let fixture: ComponentFixture<VirtualScrollWithAppendOnly>;
+    let testComponent: VirtualScrollWithAppendOnly;
+    let viewport: CdkVirtualScrollViewport;
+
+    beforeEach(waitForAsync(() => {
+      TestBed.configureTestingModule({
+        imports: [ScrollingModule, CommonModule],
+        declarations: [VirtualScrollWithAppendOnly],
+      }).compileComponents();
+      fixture = TestBed.createComponent(VirtualScrollWithAppendOnly);
+      testComponent = fixture.componentInstance;
+      viewport = testComponent.viewport;
+    }));
+
+    it('should override rendered range start', fakeAsync(() => {
```

Comment:
What happens if you scroll to e.g. item #300 and then back down to #10, do items 11+ get removed?

Reply:
No. At a glance, the fixed virtual scroll strategy doesn't reduce the `end` index when `start = 0`. So rendered items are never removed.

Reply:
I updated the virtual scroll viewport to persist the highest `end` index when `appendOnly` is enabled.